utils.py

```diff
@@ -279,3 +280,51 @@ def one_hot_and_flatten(array_2d_sequence, num_classes):
   x = onehot(array_2d_sequence, num_classes=num_classes)
   x = np.reshape(x, [x.shape[0], -1])  # flatten
   return x
+
+
+def get_mutation_pair_from_tensor_index(tensor_index: Iterable) -> Tuple[Tuple[int, int], Tuple[int, int]]:
```

Comment:
are you running pytype? I'm surprised this isn't failing. You can't do random access into an Iterable. You should use Sequence, or list.

Reply:
wow, clutch discovery, my pytype was turned off in vscode. thanks for catching!